### PR TITLE
Fixes #12986: Add a generic method to test variable existence

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/12986

### CODE REVIEW

File: tree/30_generic_methods/condition_from_variable_existence.cf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            scope => "namespace";
+
+      "${ConditionPrefix}_false"  not => isvariable("${VariableName}"),
+            scope => "namespace";
+
+  methods:
+      "success" usebundle => _classes_success("${old_class_prefix}");
+      "success" usebundle => _classes_success("${class_prefix}");
+
+      "report"  usebundle => _log_v3("Create the condition ${ConditionPrefix}_{true,false} from the existence of variable ${ConditionPrefix}", "${ConditionPrefix}", "${old_class_prefix}", "${class_prefix}", @{args});
```
Comment (ncharles) [inline tree/30_generic_methods/condition_from_variable_existence.cf]: there is no log v3 in Rudder 4.1

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

File: tree/30_generic_methods/condition_from_variable_existence.cf [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#####################################################################################
+# Copyright 2015 Normation SAS
```
Comment (amousset) [inline tree/30_generic_methods/condition_from_variable_existence.cf]: 2018

**Comment by Fdall** [comment]
PR replaced by https://github.com/Normation/ncf/pull/800

**Pull request closed without merging**